From patchwork Tue Sep  6 19:48:54 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968083
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id C665AECAAA1
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 19:49:49 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 917AF8000C; Tue,  6 Sep 2022 15:49:46 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 8A0A480008; Tue,  6 Sep 2022 15:49:46 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 5BBB88000C; Tue,  6 Sep 2022 15:49:46 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0014.hostedemail.com
 [216.40.44.14])
	by kanga.kvack.org (Postfix) with ESMTP id 3FBF080008
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 15:49:46 -0400 (EDT)
Received: from smtpin05.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay09.hostedemail.com (Postfix) with ESMTP id 23E4480867
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:46 +0000 (UTC)
X-FDA: 79882700772.05.33FB604
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf29.hostedemail.com (Postfix) with ESMTP id AC5E81200B4
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:45 +0000 (UTC)
Received: from pps.filterd (m0246617.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286IdYZg029954;
	Tue, 6 Sep 2022 19:49:42 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=pBApEGvbVXF92BIwFg9Brlkrk8UlVdTzk7cQpqnTWws=;
 b=SqUx3FL7AokuAfGoRVjI85MTQOH7QE4LJOhmxuN5AagO81FPJsNHM3tz+hAI2njOTHSr
 s1w3djpGA9PMarSxg9NsvBz2thLJdmbO4pwoSn9Kq84/IUlaUeB/myufuAs5y1+Zri2g
 BvKK6UfOYJpf2dbXWKhciRM1PzRvZvJfWsspy6QIhNxeN6bpA3uOQjzGYW93y9w1zAb3
 1avR/LxeIBu+tpbhfkNJdvLXKWskrCbUuFWy/yyYXbky9oe0kIHsTL1UFtauhu1ATsRC
 V+j1vuC+CLrWQhx9deH9AycN82rXUiXxBXUY6CEIHPILCpRcN/3pPIm71UQI4d+Hqtpy /w==
Received: from iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com
 (iadpaimrmta03.appoci.oracle.com [130.35.103.27])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbyftpvcw-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:41 +0000
Received: from pps.filterd
 (iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com [127.0.0.1])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286Je5WP039711;
	Tue, 6 Sep 2022 19:49:40 GMT
Received: from nam10-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam10lp2108.outbound.protection.outlook.com [104.47.58.108])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc9m7j9-2
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:40 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=YxhToB3539Y2gegTW9frDfcbIufBor4Mai13JybGf1+VoivZbyzhtZHJGDTUMoQURylkG8ekjyuTe0KBZK5ilkKIu7bT+/ZyFzj5eCurYyA2GVuxISk60i5t65dLK/kVCTiw5kMR/vnCtzsTQ9rRD4QFBXUAGURlX9esE8dtmYWuQ5csomqf7Z2WdnLUseisR9M6RGKNzVDPrnVrXC2kw16IzSHgXDUmqXbKI4s4UNMBWfYuEXLjEIvdR+YZz0qe1DqGGhE/lTAUuTF+pvQ9e0+RjS5/Zd23Ty5qd4sTD6puSgPMhqB2pqXL6uEdH/lDeZzwsQ7ID/6M8yAFLe9bgQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=pBApEGvbVXF92BIwFg9Brlkrk8UlVdTzk7cQpqnTWws=;
 b=XGmLOTaXREcw4EcfqkeaGqMaiBl5aZh2eFn/jhJJNLLtOEXf3hkWYdX4FKPnr2Hsfu93Nk3xW2rfmvW32Na2Upl4EmRa4O9oG8kd8OF/A1uPc1en1gGTOFn45tEMh90hfik7BZ76uOo3+rpwkhUApKzn5YuHMSvkzjFzV2mmGRcGtMUcIqILdXaMcb3e46lcLkOqJh+6DWPtcPx6O7M8DjvZBrr1hD9uDJxLscyhAa7uO8zr05eYczXS7/4dv27Eh1dpDMuR0Q0TTS88qd058f7MPi2w20M1Vh9Oxfzwa/hyTGK6Ar9+MVotOQWUz9ROal4TO0IGePNFPAHDLDC6aw==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=pBApEGvbVXF92BIwFg9Brlkrk8UlVdTzk7cQpqnTWws=;
 b=ak6Kz6HIYuhViWQ2M9Nsoh+2UMyRDIdYUBpy6ckIYJxiVsMVQk1qEbAaaeBkz63P+K8LUOaAeM+07giZkOjGi1v1jaabBfmSLVvjAK2pE5fVAk/fAVUfkPSigEnyQKAQK9P9KQCLNn46nOznCTgQhJBOlVOcjYTWt0WFj/Bz3/0=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by SJ0PR10MB5598.namprd10.prod.outlook.com (2603:10b6:a03:3d9::18) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.10; Tue, 6 Sep
 2022 19:49:36 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:49:36 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: "Matthew Wilcox (Oracle)" <willy@infradead.org>,
        Liam Howlett
	<liam.howlett@oracle.com>,
        Vlastimil Babka <vbabka@suse.cz>, Davidlohr Bueso
	<dave@stgolabs.net>
Subject: [PATCH v14 34/70] s390: remove vma linked list walks
Thread-Topic: [PATCH v14 34/70] s390: remove vma linked list walks
Thread-Index: AQHYwimx7yHW9WZ7nkWwMOHVsWp71g==
Date: Tue, 6 Sep 2022 19:48:54 +0000
Message-ID: <20220906194824.2110408-35-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 1640c5a2-1893-4a2a-9c63-08da9040ed77
x-ms-traffictypediagnostic: SJ0PR10MB5598:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 0M7+OI3EYiTA+YTM5f2KrYNdWa1QfoLi/umK0joNs/Xyl4x4Xn4a3Q1H52IB1SSgmjv/YWb5g9RssLwXUd7Ju2V4YAl5gXUk4TFgHG2XuhtmIfLMespbaN9cwRUorNu6uo/IIN1eOxt1X2x4kNTOTN6uZXFSRSgLa2zxlXv7qWkRt+MTepGq1NlfIXuyjKQj+EbumO4bf/EE6F7MJQ1z34xQFWd64jFXjWwaZpAk0D6N1iVME2rJHNxJyteqPDIftQyBlm8nu75SPzt/ZHMbpWyoh0CWthesG+Aa3p7ius/4w2fmSRko4niY07rPTn/crMXLdgfclWIPaR64tUFTsCcSurfiKOpHWl1tYtBSmw+1QDCPwTbKqWQwi9M0NA+r7YvJOSiPSHGsuZTLRgkmC499saHy2ZzKJN5LI/fopYnlWVVlGrfWVTo+38CnWWs7ofYTzbZSSj0hneBBdVNolpH8fp6BfxWoAhe2LlWS4yKSrtdl7QOzWXP0qfUAaLNI20NNz4NKW1wdZqPjb+NZyyqEZ4keswXFcyMTOtFiPBXE808cwHsrKl0KwujiCnKFdCAv70zJy4ILYsvpkp8q4p+Y01zX/R4rAtypNpH2e7k8mL0fLpNAcQIDZ9bbA+Hp3OLEwmRyxLvqkBrLimBXK7tQu03Smvo8nnKnr8KS6oaQMq12f4J4cLALsPLdL5ftU+Q9rDFU+zicZdIUmkKchaTpTz0uIAuLSeM/z0RChLOrhELUtZi4++ba67QTTefVwU3Wgq/0uOD+UeNJvrXOrQ==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(346002)(39860400002)(396003)(136003)(376002)(366004)(64756008)(66556008)(41300700001)(91956017)(76116006)(66476007)(66446008)(66946007)(71200400001)(54906003)(110136005)(36756003)(316002)(6512007)(44832011)(8676002)(122000001)(2906002)(6486002)(4326008)(186003)(8936002)(5660300002)(478600001)(38070700005)(38100700002)(2616005)(86362001)(26005)(1076003)(6666004)(6506007)(83380400001);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?uX4L4+O0jHtJliDsmeTY5Wg?=
	=?iso-8859-1?q?fWUx+oUrhe3M+isVoXMy/gFmkqMMR/5OFdUoQc4Ew/5Xao/O9CU6qtX2eP5w?=
	=?iso-8859-1?q?LosB1ObfqDxq/5h1DIxa7O2z9wpGxiQQyfqb5Fpd6odivgfyeiqvAEKFg7ih?=
	=?iso-8859-1?q?q6VRmBorBWt0vFF+ypLWaHNEflO9dGbAFHYlC76z+j1kWZiZkaXHMu4Zfik4?=
	=?iso-8859-1?q?i21imfStn9IADyy+aYhiNhkUIeKlvx7Qwo0/MA9ox1iQcl0gcaL/wDE1HHNB?=
	=?iso-8859-1?q?3PwI3KL9Gy7KQfPXzL4yWpbXDVJ+2UStWjv3w+0mQE3AuozMvZoRFxDvl0MJ?=
	=?iso-8859-1?q?LL5FXcFOAmhPfLAomoLIPcI8tF6uR7J4Owu6LTUfup6xprYK0Bn626xZxnj1?=
	=?iso-8859-1?q?fe7dYDDY+F5ZescxCbuXrCOQTDFnDc+u63jfI7c+VclVUIX+nLNQ6kdkMD18?=
	=?iso-8859-1?q?0UYTefSbw4tnOj5WzG4IiA3n0NppwK6h2puig69KhL1F8hs5w5IAFetBAgey?=
	=?iso-8859-1?q?FaSia48EMUU5cul9elopguJHbvwHwLiGZ9TusmSOOmEXfXlijZ1FoefDCUjt?=
	=?iso-8859-1?q?8YQwFmWq1NEyV9kfoiQQO7ctqz4Nh7XeoBr5fRgaV4Cze0EC2RHJ3PzWN0T9?=
	=?iso-8859-1?q?czNn9qMNB+sWXlWe9JNOK7O2NCZtJ+C/PQsM93EiT+jWgLZzVIzsJzXvTABk?=
	=?iso-8859-1?q?vfkx97XrzmqYJbfKdOVbDHBThXALdEk5ciXUIeCjxdkK6BgheGoPZQZYS0Is?=
	=?iso-8859-1?q?Ap99jtGyWxuhGqEQ/N62J0h1RC07HhQZGHWCV/GUiy29sUMKx+BZ3m76bcme?=
	=?iso-8859-1?q?HWzA/MB91GkevNkQzuBSuhl1+dy2GLyhmlZlLiZOqwTw7wlfk5iJN8jZxwXu?=
	=?iso-8859-1?q?5NS71U4Wf+GCCfF8+66PmZpGWXy5htTAQ+Ey3MqiSRmsNJCQVxsPWRu63eV8?=
	=?iso-8859-1?q?/YRS6Ddc3Qzx8EXXrJUyYSAPpgRfglr6yqv/Xw+uzVhsFS6fDr0+dny0GUOQ?=
	=?iso-8859-1?q?f/Y7BgTxr1UuYEeED5wFK2nN7m7g4GLQ96Xgh/OTmZpXgJR7v8FpR3shz+LV?=
	=?iso-8859-1?q?RyOb10h5rxumFX3Xxi08k+SQQttUopPCinbvbZa+AdWGKG7QBHhpJmSDmCOM?=
	=?iso-8859-1?q?lhKHgPpbzkqB+6N+akX+4Xc4pa2iK2vxc9vdIqAdhOMv+zV+jwwDAkaCqAff?=
	=?iso-8859-1?q?OKNh5HaYNyYjgSn2zKHYeKhraZ+Jw/yhhPD9kXwpuPohPNTutSWMy5mc8QGZ?=
	=?iso-8859-1?q?ku+veqX52Ejj4a6d+53zdJ0uvzDuw3JmjOCAFhY9mHSkMJFp4ABg2NxplrRu?=
	=?iso-8859-1?q?yGMAXsoPEWV8FVsh9AyT4kesqSudjLmm3PJZivYmS9Zf+/pZVnU+l3M4e7H+?=
	=?iso-8859-1?q?T5KLFw4VLu/av45wzQtDSq6GNiRaILy9Mkkuq/fi5g2sEf/SuT4ANtWPy//i?=
	=?iso-8859-1?q?eGwkD8ejNSYaEbWyv633kcFcPRtJnOrBH1mCY/n+ZzUcMm/8921Ag9pmd82t?=
	=?iso-8859-1?q?NkGEM4ADgaSIXKN5G+4JPYzGk+qVXraljsHPHMQn01K3nSLWbyYhDe4Rep9L?=
	=?iso-8859-1?q?kg2MkJMaYIxb3ZlgE8pNpEdH1sjLatLmg2Q5d/Wz9vZm6ej3oYFFhyj+ZVVJ?=
	=?iso-8859-1?q?0UXCDaAqXYsHDCmMWtqE/TP2ezHGjIaEKU3VDyg=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 1640c5a2-1893-4a2a-9c63-08da9040ed77
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:48:54.3970
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 2HMkNNlV8QPzpy3fGoFb0Ph0nPGZG/neQulByn7BdMzu//t1XLHB5Hy8KPXXFW8D3D0WptxRHZpSGEG24TqiTw==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SJ0PR10MB5598
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 bulkscore=0
 mlxlogscore=999
 phishscore=0 mlxscore=0 spamscore=0 malwarescore=0 suspectscore=0
 adultscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.12.0-2207270000 definitions=main-2209060091
X-Proofpoint-ORIG-GUID: I5xJVudibNZASZFBum75hpiVqxeSBqHi
X-Proofpoint-GUID: I5xJVudibNZASZFBum75hpiVqxeSBqHi
ARC-Authentication-Results: i=2;
	imf29.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=SqUx3FL7;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=ak6Kz6HI;
	spf=pass (imf29.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	dmarc=pass (policy=none) header.from=oracle.com
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662493785; a=rsa-sha256;
	cv=pass;
	b=jfFrrf7oNCYrcNr9KN0jFRq76qrJ0ncB/ti+JqEcEBpx4W7pshJZNjoAEL6Zf73Ei/HyDy
	SoWw3zwJCJLsRoqCIkOH2O6mXlxmLBLg10/AgB+71GnfG2mXHW5Ndib9RMakWogx+O8W2+
	LJgFSsGgQ6QPPpMZbSK7gVay/qLyaR4=
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662493785;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=pBApEGvbVXF92BIwFg9Brlkrk8UlVdTzk7cQpqnTWws=;
	b=CGNzjGtxBxRiO4jxzSoqfsO4jEizN4S718zchA0AwYN2xn+AxtGS1eLVreH3IaY8006rlm
	MjPSBQp00Fcg6aORgLN4TU7HP/zXDehoeJVL7crbJiZMBg0RqAoqg37wVv0q97Dzfot8vi
	OLl3fgpk2tMMLNn+IO6JOoifotK1NHs=
X-Rspamd-Server: rspam04
X-Rspamd-Queue-Id: AC5E81200B4
X-Rspam-User: 
Authentication-Results: imf29.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=SqUx3FL7;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=ak6Kz6HI;
	spf=pass (imf29.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	dmarc=pass (policy=none) header.from=oracle.com
X-Stat-Signature: w3d581o16zpnioft9a1k1p4xmcncxmtn
X-HE-Tag: 1662493785-710343
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Matthew Wilcox (Oracle)" <willy@infradead.org>

Use the VMA iterator instead.

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
Reviewed-by: Davidlohr Bueso <dave@stgolabs.net>
---
 arch/s390/kernel/vdso.c | 3 ++-
 arch/s390/mm/gmap.c     | 6 ++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/s390/kernel/vdso.c b/arch/s390/kernel/vdso.c
index 5075cde77b29..535099f2736d 100644
--- a/arch/s390/kernel/vdso.c
+++ b/arch/s390/kernel/vdso.c
@@ -69,10 +69,11 @@ static struct page *find_timens_vvar_page(struct vm_area_struct *vma)
 int vdso_join_timens(struct task_struct *task, struct time_namespace *ns)
 {
 	struct mm_struct *mm = task->mm;
+	VMA_ITERATOR(vmi, mm, 0);
 	struct vm_area_struct *vma;
 
 	mmap_read_lock(mm);
-	for (vma = mm->mmap; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		unsigned long size = vma->vm_end - vma->vm_start;
 
 		if (!vma_is_special_mapping(vma, &vvar_mapping))
diff --git a/arch/s390/mm/gmap.c b/arch/s390/mm/gmap.c
index 62758cb5872f..02d15c8dc92e 100644
--- a/arch/s390/mm/gmap.c
+++ b/arch/s390/mm/gmap.c
@@ -2515,8 +2515,9 @@ static const struct mm_walk_ops thp_split_walk_ops = {
 static inline void thp_split_mm(struct mm_struct *mm)
 {
 	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, 0);
 
-	for (vma = mm->mmap; vma != NULL; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		vma->vm_flags &= ~VM_HUGEPAGE;
 		vma->vm_flags |= VM_NOHUGEPAGE;
 		walk_page_vma(vma, &thp_split_walk_ops, NULL);
@@ -2584,8 +2585,9 @@ int gmap_mark_unmergeable(void)
 	struct mm_struct *mm = current->mm;
 	struct vm_area_struct *vma;
 	int ret;
+	VMA_ITERATOR(vmi, mm, 0);
 
-	for (vma = mm->mmap; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		ret = ksm_madvise(vma, vma->vm_start, vma->vm_end,
 				  MADV_UNMERGEABLE, &vma->vm_flags);
 		if (ret)

